src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+interface Props {
+  todo: Todo;
+  changeModal: (id: number) => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ todo, changeModal }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | undefined>();
```

Comment:
Consider initializing `user` with `null` instead of `undefined` to better indicate the absence of a user initially. This can help avoid potential confusion or errors when checking the user's existence.